qmcpy/algorithms/function/asian_call_integrand.py

```diff
@@ -0,0 +1,50 @@
+''' Originally developed in MATLAB by Fred Hickernell. Translated to python by Sou-Cheng T. Choi and Aleksei Sorokin '''
+import numpy as np
+
+from algorithms.function.integrand_base import IntegrandBase
+
+
+class AsianCallFun(IntegrandBase):
+    """
+    Specify and generate payoff values of an Asian Call option
+
+    TODO - Come up with better, more interpretable, lowercase names for these parameters
+
+    """
+    def __init__(self, bm_measure=None, volatility=.5, S0=30, K=25, nominal_value=None):
+        super().__init__(nominal_value)
+        self.bm_measure = bm_measure
+        self.volatility = volatility
+        self.S0 = S0
+        self.K = K
+        self.dimFac = 0
+
+        if self.bm_measure:
+            num_bm = len(bm_measure)
+            self.fun_list = [AsianCallFun() for i in range(num_bm)]
```

Comment:
This just looks so weird -- recursion in Python is generally frowned upon (though permitted) but creating a class populated by itself is super odd.  My suggestion, eventually, would be to just create a subclass called `AsianCallFunWithoutBM`, or maybe something better than that.

Reply:
I agree that it looks strange. 
It's a little bit of a hacky way to allow for mulit-dimensional problems and to hide their construction from the user. 
If there is an easier way to implement something like this that may be helpful to us and users. 
Note: there are similar constructions for Measure and DiscreteDistributuion.